compiler/rustc_metadata/src/native_libs.rs

```diff
@@ -95,14 +95,15 @@ pub fn try_find_native_static_library(
     name: &str,
     verbatim: bool,
 ) -> Option<PathBuf> {
+    let target = (&*sess.target.staticlib_prefix, &*sess.target.staticlib_suffix);
+    let unix = ("lib", ".a");
     let formats = if verbatim {
-        vec![("".into(), "".into())]
+        vec![("", "")]
+    } else if target != unix && sess.target.is_like_msvc {
+        // On Windows MSVC naming scheme `libfoo.a` is used as a fallback from default `foo.lib`.
```

Comment:
I'm not sure this is correct. To my knowledge, MSVC does not use `.a` at all for static libs (unless we/the Rust ecosystem have started doing this somewhere). What probably does use `.a` though would be `*-windows-gnu`. 

This code originated a long time ago, far before MSVC was supported: cb823b09dfc2ca71bda66e874c5e450038fbec1d. My hunch is that this probably needs to either be changed to target Windows generally or `*-windows-gnu` more specifically.

Reply:
All built-in targets except windows-msvc already have the `libname.a` naming scheme in their target specs, so this fallback doesn't apply to them (due to the `target != unix` condition). So on windows-gnu in particular the fallback is also not necessary.

>unless we/the Rust ecosystem have started doing this somewhere

Meson is using the `libname.a` naming scheme on MSVC and CMake now also supports it due to Meson (#123436, https://mesonbuild.com/FAQ.html#why-does-building-my-project-with-msvc-output-static-libraries-called-libfooa, https://gitlab.kitware.com/cmake/cmake/-/issues/23975).

Reply:
Ok, can you mention that Meson and CMake do this in the comment?